plugins/feeds/public/abusech_malwarebazaar.py

```diff
@@ -63,14 +63,22 @@ def analyze(self, block):  # pylint: disable=arguments-differ
         context["imphash"] = block["hash"]["imphash"]
         context["ssdeep"] = block["hash"]["ssdeep"]
 
-        if block.get("hash", []):
+        if block.get("sha256_hash", ""):
+            malware_file = File.get_or_create(
+                value='FILE:{}'.format(block["sha256_hash"]))
+
+            malware_file.add_context(context)
+            malware_file.tag(block["file_type_mime"])
             # md5, sha1, sha256
-            for hash_type in ("md5_hash", "sha1_hash", "sha256_hash"):
+            for hash_type in ("md5_hash", "sha1_hash", "sha256_hash") or []:
                 try:
                     hash_data = Hash.get_or_create(value=block[hash_type])
                     hash_data.add_context(context)
                     hash_data.add_source(self.name)
                     if block.get("tags"):
                         hash_data.tag(block["tags"])
+
+                    malware_file.active_link_to(
+                        hash_data, hash_data.split("_")[0], self.name)
```

Comment:
You probably mean `hash_type` instead of `hash_data` here

Reply:
ah yes bad cp

Reply:
pushed update